### PR TITLE
feat: login to download

### LINKED ISSUE #2554: feat: require log in to download on platform (how-to)
**Is your feature request related to a problem? Please describe.**
We want to encourage users to collaborate on the platform / understand who is downloading the files that accompany research and how to items.

**Describe the solution you'd like**
Download buttons should direct to sign in like how we redirect for the useful and follow buttons.

### PR DESCRIPTION
**Pull request by iSCJT**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Adds redirectToSignIn property to file download components and configures it for how-to's.  If logged out the download link is removed and the button redirects to the sign in page.

## Git Issues

Closes #2554

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Review by evakill — APPROVED** [review]
this diff looks good!

imo the names of these components are kinda confusing. if we wanted to do cleanup here i'd recommend something like `FileInformation` -> `DonwloadStaticFile` and `DownloadFiles` -> `DownloadFileFromLink`

File: packages/components/src/DownloadFiles/DownloadFiles.tsx [hunk 1]
```diff
@@ -25,12 +27,34 @@ export const DownloadFiles = (props: DownloadFilesProps) => {
           maxWidth: '300px',
           borderRadius: 1,
         }}
+        onClick={() =>
+          props.redirectToSignIn ? history.push('/sign-in') : undefined
```
Comment (thisislawatts) [inline packages/components/src/DownloadFiles/DownloadFiles.tsx:31]: **issue:** Do we want to introduce awareness of `react-router-dom` to the component library? This seems like an external concern for the consuming library. For example if we wanted to use this component within a serverless function would it work.

Comment (iSCJT) [reply packages/components/src/DownloadFiles/DownloadFiles.tsx]: Good point, I've moved that out.

**Comment by iSCJT** [comment]
@evakill Agree the naming was confusing, I've refactored to the naming you suggested.

**Review by evakill — APPROVED** [review]
thanks for the update 👌

**Pull request merged**